### PR TITLE
lib: Add a standard parser option for JSON formatting

### PR DESCRIPTION
**Pull request by kritibirda26**
As a part of https://github.com/OSGeo/grass/discussions/3019, JSON format support will be added to multiple modules. By default, modules output in existing plain format. If the format=json option is provided, modules will output in JSON format instead. To avoid duplication of code across several modules, define a standard parser option.

### CODE REVIEW

File: lib/gis/parser_standard_options.c
Comment (petrasovaa) [inline lib/gis/parser_standard_options.c:954]:
```suggestion
        Opt->descriptions = _("plain;Plain text output;"
```

**Review by petrasovaa — CHANGES_REQUESTED** [review]
Please also replace it in v.db.select, the options are different there, so you keep the current v.db.select options and descriptions.

I would think the format should be by default required, although since it has default answer, it probably doesn't matter that much.

**Comment by wenzeslaus** [comment]
This is great improvement @kritibirda26. To fix the OSGeo4W CI, please update your branch from main (e.g. Update branch on GitHub or merge main locally into this branch).

File: lib/gis/parser_standard_options.c
Comment (kritibirda26) [reply lib/gis/parser_standard_options.c]: Fixed.

**Comment by kritibirda26** [comment]
@petrasovaa Hi! I have updated v.db.select and also changed the option to be required by default.

**Comment by kritibirda26** [comment]
The tests are failing with this weird error but the build passes locally. Also, the string that is said to be invalid is actually a valid isoformat string. Any suggestions on how to debug?

```
gnu/docs/html/g.gui.timeline.html
Traceback (most recent call last):
  File "/home/runner/work/grass/grass/dist.x86_64-pc-linux-gnu/utils/mkhtml.py", line 921, in <module>
    git_commit = get_last_git_commit(
  File "/home/runner/work/grass/grass/dist.x86_64-pc-linux-gnu/utils/mkhtml.py", line 397, in get_last_git_commit
    return parse_git_commit(
  File "/home/runner/work/grass/grass/dist.x86_64-pc-linux-gnu/utils/mkhtml.py", line 242, in parse_git_commit
    git_log["date"] = format_git_commit_date_from_local_git(
  File "/home/runner/work/grass/grass/dist.x86_64-pc-linux-gnu/utils/mkhtml.py", line 345, in format_git_commit_date_from_local_git
    return datetime.fromisoformat(
ValueError: Invalid isoformat string: '2024-05-17T16:09:45Z'
```

https://github.com/OSGeo/grass/actions/runs/9131120968/job/25109628154?pr=3704#step:8:6469

**Comment by neteler** [comment]
A guess: it probably expects a string without the 'Z' character?

**Comment by kritibirda26** [comment]
@neteler I see, do you know if there is a local git setting I can configure to fix the datetime format for the commit timestamp?

**Comment by wenzeslaus** [comment]
This is failing across several PRs with repeated runs. I'm opening a separate issue to discuss this.

**Comment by echoix** [comment]
What prefix for the title of the PR should be used before merging, once it is rebased? (the build issue is most probably fixed now on main).

File: raster/r.horizon/main.c
Comment (petrasovaa) [inline raster/r.horizon/main.c:302]:
```suggestion
```

File: raster/r.kappa/main.c
Comment (petrasovaa) [inline raster/r.kappa/main.c:101]:
```suggestion
```

File: raster/r.what/main.c
Comment (petrasovaa) [inline raster/r.what/main.c:146]: The default is required = YES now, so this is technically changing the behavior, but I think that's fine.

File: include/grass/gis.h
Comment (nilason) [inline include/grass/gis.h:357]:
```suggestion
    G_OPT_F_FORMAT, /*!< set output format to plain text or JSON */
```

File: raster/r.what/main.c
Comment (kritibirda26) [reply raster/r.what/main.c:146]: Makes sense.

**Comment by echoix** [comment]
@petrasovaa previously requested changes, she might want to place an approval to overrule her review.

And a prefix for the PR, so it can be classified in the release notes. I'm not sure which one to apply.

**Review by petrasovaa — APPROVED** [review]

**Comment by ninsbl** [comment]
Sorry for beeing late here.

I just noticed that this new standard parser option:
1) is not groupt with the other `F` options but comes last in the list (so out of order). See: https://grass.osgeo.org/grass84/manuals/parser_standard_options.html 
2) does not have a decription (seems to be a typo with description**s**)
3) should probably explicitly default to NO for "multiple"

Should the format option also cover other formats (CSV, shell, ...), those are currently missing, but a long list of possible formats will probably defeat the purpose of the standard paser option?

Just comments you may consider... Anyway, I am glad to see this standard parser option...

**Pull request merged**
